js/TableDetailComponent.js

```diff
@@ -88,7 +92,7 @@ class TableDetailComponent {
 
         // Close ESC button
         this.$html.on('keydown', (event) => {
-            if (event.keyCode === 27) {
+            if (event.keyCode === 27 && this.$tableDetailBackdrop.hasClass('in')) {
```

Comment:
Really minor here I'm not suggesting we change it unless you feel really compelled to do so. In instances like this it might be nicer to determine if something is active by accessing some internal state in our Component, like an isActive getter or something. The problem with hanging off the presentation class "in" is aside from mixing our concerns we can not change the implementation of "in" now without having to factor this service also. 

That might feel like a pain if we assumed we were making a purely presentational (HTML & CSS) change like changing "in" to "is-in" (as a contrived example), but now we also need to change javascript components and their tests.

Reply:
@mikedevelops Updated to do this. Totally agree about separating concerns.